great_expectations/rule_based_profiler/config/base.py

```diff
@@ -46,28 +45,16 @@ def make_config(self, data: dict, **kwargs) -> Type[DictDot]:
                 "The subclass extending NotNullSchema must define its own custom __config_class__"
             )
 
-        # Removing **kwargs before creating config object
-        recognized_attrs = {f.name for f in dataclasses.fields(self.__config_class__)}
-        cleaned_data = filter_properties_dict(
-            properties=data,
-            keep_fields=recognized_attrs,
-            clean_nulls=False,
-            clean_falsy=False,
-        )
-
-        return self.__config_class__(**cleaned_data)
```

Comment:
No more dataclasses in this implementation

Reply:
The dataclass constructor was somehow messing with the subclass' relationship with the parent - keeping current style of implementation for the time being